docs/extending/query.md

```diff
@@ -1,197 +1,209 @@
 
-- `is_collection` (optional, default `false`): A boolean indicating whether the response data is a collection. If false, only a single item will be returned.
-- `mappings` (required): An associative array of output variable definitions. The keys of the array are machine-friendly output variable names and the values are associative arrays with the following structure:
-  - `name` (optional): The human-friendly display name of the output variable.
-  - `default_value` (optional): The default value for the output variable.
-  - `path` (required): A [JSONPath](https://jsonpath.com/) expression to extract the variable value.
-  - `type` (required): The type of the output variable. Supported types are -
-    - `id`
-    - `base64`
-    - `boolean`
-    - `number`
-    - `string`
-    - `button_url`
-    - `image_url`
-    - `image_alt`
-    - `currency`
-    - `markdown`
+- `name` (optional): The human-friendly display name of the input variable
+- `default_value` (optional): The default value for the input variable.
+- `type` (required): The primitive type of the input variable. Supported types are:
+  - `boolean`
+  - `id`
+  - `integer`
+  - `null`
+  - `number`
+  - `string`
 
 #### Example
 
 ```php
-public function get_output_schema(): array {
-	return [
-		'is_collection' => false,
-		'mappings'      => [
-			'zip_code' => [
-				'name' => 'Zip Code',
-				'path' => '$["post code"]',
-				'type' => 'string',
-			],
-			'city'     => [
-				'name' => 'City',
-				'path' => '$.places[0]["place name"]',
-				'type' => 'string',
-			],
-			'state'    => [
-				'name' => 'State',
-				'path' => '$.places[0].state',
-				'type' => 'string',
-			],
-		],
-	];
-}
+'input_schema' => [
+	'zip_code' => [
+		'name' => 'Zip Code',
+		'type' => 'string',
+	],
+],
 ```
 
-The default implementation returns an empty array.
+If omitted, it defaults to an empty array.
 
-### get_data_source(): DataSourceInterface
+### output_schema: array (required)
 
-The `get_data_source` method returns the data source associated with the query. By default, this method returns the data source that was provided to the class constructor. In most instances, you should not need to override this method.
+The `output_schema` property defines how to extract data from the API response. The method should return an associative array with the following structure:
 
-### get_endpoint( array $input_variables ): string
-
-By default, the `get_endpoint` method proxies to the `get_endpoint` method of query's data source. Override this method to set a custom endpoint for the query—for example, to construct the endpoints using an input variable. The input variables for the current request are provided as an associative array (`[ $var_name => $value ]`).
+- `format` (optional): A callable function that formats the output variable value.
+- `generate` (optional): A callable function that generates or extracts the output variable value from the response, as an alternative to `path`.
```

Comment:
I see that this is an alternative to `path`, but `path` below is listed as (required). A bit confusing, clarification here would be nice.

Reply:
Addressed in https://github.com/Automattic/remote-data-blocks/pull/243/commits/5060870e147da1b4895e26f87c1fea67b1d3d110